### PR TITLE
Added the StyleCop.Analyzers NuGet to enforce styles

### PR DESCRIPTION
**Pull request by mattleibow**
- at this time there is no way to enforce the .editorconfig
 - we will have to duplicate the rules because the IDE uses the .editorconfig for some things and StyleCop adds some more

StyleCop is very powerful and can be used right now. The build task for enforcing .editorconfig is in development: https://github.com/dotnet/roslyn/projects/18

### CODE REVIEW

File: Caboodle.ruleset [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    <Rule Id="SA1132" Action="Warning" />
+    <Rule Id="SA1133" Action="Warning" />
+    <Rule Id="SA1134" Action="Warning" />
+    <Rule Id="SA1135" Action="Warning" />
+    <Rule Id="SA1136" Action="Warning" />
+    <Rule Id="SA1137" Action="Warning" />
+    <Rule Id="SA1139" Action="Warning" />
+    <Rule Id="SA1200" Action="Warning" />
+    <Rule Id="SA1201" Action="Warning" /><!-- TODO: verify SA1201ElementsMustAppearInTheCorrectOrder -->
+    <Rule Id="SA1202" Action="Warning" /><!-- TODO: verify SA1202ElementsMustBeOrderedByAccess -->
```
Comment (mattleibow) [inline Caboodle.ruleset]: Do we care about element ordering?

File: Caboodle.ruleset [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    <Rule Id="SA1412" Action="Warning" />
+    <Rule Id="SA1413" Action="None" />
+    <Rule Id="SA1500" Action="Warning" />
+    <Rule Id="SA1501" Action="Warning" />
+    <Rule Id="SA1502" Action="Warning" />
+    <Rule Id="SA1503" Action="None" />
+    <Rule Id="SA1504" Action="Warning" />
+    <Rule Id="SA1505" Action="Warning" />
+    <Rule Id="SA1506" Action="Warning" />
+    <Rule Id="SA1507" Action="Warning" /><!-- TODO: verify SA1507CodeMustNotContainMultipleBlankLinesInARow -->
```
Comment (mattleibow) [inline Caboodle.ruleset]: Do we care about multiple lines of open-ness?

File: Caboodle.ruleset [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    <Rule Id="SA1640" Action="None" />
+    <Rule Id="SA1641" Action="None" />
+    <Rule Id="SA1642" Action="None" />
+    <Rule Id="SA1643" Action="None" />
+    <Rule Id="SA1644" Action="None" />
+    <Rule Id="SA1645" Action="None" />
+    <Rule Id="SA1646" Action="None" />
+    <Rule Id="SA1647" Action="None" />
+    <Rule Id="SA1648" Action="None" />
+    <Rule Id="SA1649" Action="Warning" /><!-- TODO: verify SA1649FileNameMustMatchTypeName -->
```
Comment (mattleibow) [inline Caboodle.ruleset]: Must out class names and file names match?

Comment (jamesmontemagno) [reply Caboodle.ruleset]: No

File: Caboodle.ruleset (hunk 2 above)
Comment (jamesmontemagno) [reply Caboodle.ruleset]: probably not

File: Caboodle.ruleset (hunk 1 above)
Comment (jamesmontemagno) [reply Caboodle.ruleset]: probably not

**Comment by mattleibow** [comment]
build

**Pull request merged**